pkg/cluster/majorversionupgrade.go

```diff
@@ -54,6 +58,58 @@ func (c *Cluster) isUpgradeAllowedForTeam(owningTeam string) bool {
 	return util.SliceContains(allowedTeams, owningTeam)
 }
 
+func (c *Cluster) PatchAnnotations(patchBytesAnnotation []byte) error {
+	resourceName := c.Name
+	_, err := c.KubeClient.Postgresqls(c.Namespace).Patch(context.Background(), resourceName, types.MergePatchType, patchBytesAnnotation, metav1.PatchOptions{})
+	if err != nil {
+		c.logger.Errorf("failed to patch annotations: %v", err)
+		return err
+	}
+	return nil
+}
+
+func (c *Cluster) annotatePostgresResource(isSuccess bool) error {
+	annotations := make(map[string]string)
+	currentTime := metav1.Now().Format("2006-01-02T15:04:05Z")
+	if isSuccess {
+		annotations["last-major-upgrade-success"] = currentTime
+	} else {
+		annotations["last-major-upgrade-failure"] = currentTime
+	}
+
+	patchAnnotation := map[string]map[string]map[string]string{
+		"metadata": {
+			"annotations": annotations,
+		},
+	}
+	patchBytes, err := json.Marshal(patchAnnotation)
+	if err != nil {
+		c.logger.Errorf("failed to marshal annotations: %v", err)
+		return err
+	}
+
+	err = c.PatchAnnotations(patchBytes)
+	if err != nil {
+		c.logger.Errorf("failed to patch annotations to Postgres resource: %v", err)
+		return err
+	}
+	return nil
+}
+
+func (c *Cluster) getLastMajorUpgradeAnnotations() map[string]string {
+	annotations := c.ObjectMeta.GetAnnotations()
+	upgradeAnnotationKeys := []string{"last-major-upgrade-failure", "last-major-upgrade-success"}
+	upgradeAnnotationValues := make(map[string]string)
+	for _, annotationKey := range upgradeAnnotationKeys {
+		if val, exists := annotations[annotationKey]; exists {
+			upgradeAnnotationValues[annotationKey] = val
+		} else {
+			upgradeAnnotationValues[annotationKey] = ""
+		}
+	}
+	return upgradeAnnotationValues
+}
+
```

Comment:
do we really need this function? Can someone not just do `if _, exists := c.Spec.ObjectMeta.Annotations[annotationKey]; if exists` wherever this needs to be checked?